net-dns/dnsdist/dnsdist-1.1.0.ebuild

```diff
@@ -0,0 +1,135 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+EGIT_REPO_URI="https://github.com/PowerDNS/pdns.git"
+
+if [[ ${PV} = 9999 ]]; then
+	ADDITIONAL_ECLASSES="autotools git-r3"
+fi
+
+inherit eutils systemd user ${ADDITIONAL_ECLASSES}
+
+DESCRIPTION="A highly DNS-, DoS- and abuse-aware loadbalancer."
+HOMEPAGE="http://dnsdist.org"
+
+if [[ ${PV} == 9999 ]]; then
+	SRC_URI=""
+	S="${WORKDIR}/${P}/pdns/dnsdistdist"
+else
+	SRC_URI="https://downloads.powerdns.com/releases/${P}.tar.bz2"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="dnscrypt luajit readline regex remote-logging +ssl systemd test"
+
+RESTRICT="readline? ( bindist )"
+
+REQUIRED_USE="dnscrypt? ( ssl )"
+
+DEPEND="
+	>=dev-libs/boost-1.35:=
+	luajit? ( dev-lang/luajit:= )
+	!luajit? ( >=dev-lang/lua-5.1:= )
+	remote-logging? ( dev-libs/protobuf:= )
+	readline? ( sys-libs/readline:0= )
+	!readline? ( dev-libs/libedit:= )
+	regex? ( dev-libs/re2:= )
+	ssl? ( dev-libs/libsodium:= )
+	systemd? ( sys-apps/systemd:= )
+"
+
+RDEPEND="${DEPEND}"
+
+[[ ${PV} == 9999 ]] && DEPEND+="
+	app-text/pandoc
+	dev-util/ragel
+"
+
+src_prepare() {
+	default
+
+	[[ ${PV} == 9999 ]] && eautoreconf
+
+	if use readline ; then
+		# sed's --follow-symlinks is a must here for the 9999 version, since those files are placed in ../.
+		# in git, and then symlinked to
+
+		sed --follow-symlinks -i \
+			-e 's~^#include <editline/readline.h>$~#include <readline/readline.h>~g' dnsdist.cc \
+			|| die "dnsdist.cc: Sed broke!"
+
+		sed --follow-symlinks -i \
+			-e 's~^#include <editline/readline.h>$~#include <readline/readline.h>'"\n"'#include <readline/history.h>~g' dnsdist-console.cc \
```

Comment:
Actually, I'm not going to accept this with those seds. They're too complex, too hard to read and a recipe for future failure (remember that sed that does not match does not trigger an error). Please use a patch instead.

And since we hate conditional patching, it would probably be meaningful to add some `#ifdef` + `CPPFLAGS` appending to handle the switch between readline & libedit. If you really want, you can even patch configure to support switching between libedit & readline -- maybe even try upstreaming that, if you believe upstream could accept such a change.